lib-compose/src/main/java/com/what3words/components/compose/maps/buttons/MyLocationButton.kt

```diff
@@ -117,123 +134,165 @@ fun MyLocationButton(
+                enabled = isButtonEnabled
             ) {
                 Icon(
-                    imageVector = if (isLocationActive) Icons.Default.MyLocation else Icons.Default.LocationSearching,
+                    painter = locationIconVector,
                     contentDescription = "Location icon",
-                    tint = if (isLocationActive) Color(0xFF14B5FF) else Color.Black, // TODO: Define 0xFF14B5FF color name
+                    tint = locationIconColor,
                     modifier = Modifier.size(locationButtonConfig.locationIconSize)
                 )
             }
             if (accuracyDistance >= SAFE_ACCURACY_DISTANCE) {
-                Box(
-                    modifier = Modifier
-                        .offset(x = 4.dp, y = (-4).dp)
-                        .size(locationButtonConfig.accuracyIndicatorSize)
-                        .clip(CircleShape)
-                        .background(
-                            color = when {
-                                accuracyDistance in SAFE_ACCURACY_DISTANCE until WARNING_ACCURACY_DISTANCE -> MaterialTheme.w3wColorScheme.warning
-                                accuracyDistance >= WARNING_ACCURACY_DISTANCE -> MaterialTheme.colorScheme.error
-                                else -> return
-                            }
-                        )
-                        .align(Alignment.TopEnd),
-                    contentAlignment = Alignment.Center
-                ) {
-                    Icon(
-                        imageVector = Icons.Outlined.WarningAmber,
-                        contentDescription = null,
-                        modifier = Modifier.padding(4.dp),
-                        tint = MaterialTheme.colorScheme.onError
-                    )
-                }
+                WarningIndicator(
+                    modifier = Modifier.align(Alignment.TopEnd),
+                    accuracyDistance = accuracyDistance,
+                    isDarkMode = isDarkMode,
+                    indicatorSize = locationButtonConfig.accuracyIndicatorSize,
+                )
             }
         }
     }
 }
 
-@Preview
 @Composable
-private fun A1() {
-    W3WTheme {
-        MyLocationButton(
-            modifier = Modifier,
-            onMyLocationClicked = {},
-            accuracyDistance = 0,
-            isLocationEnabled = false,
-            isLocationActive = false,
-            unitMetrics = METER
+private fun WarningIndicator(
+    modifier: Modifier,
+    accuracyDistance: Int,
+    isDarkMode: Boolean,
+    indicatorSize: Dp,
+) {
+
+    val indicatorBackgroundColor = when {
+        accuracyDistance in SAFE_ACCURACY_DISTANCE until WARNING_ACCURACY_DISTANCE -> {
+            if (isDarkMode) Color(0xFFFFDA88) else Color(0xFFF8C03C)
+        }
+
+        accuracyDistance >= WARNING_ACCURACY_DISTANCE -> Color(0xFFF26C50)
+        else -> return
+    }
+
+    val indicatorIconColor = if (isDarkMode) Color(0xFF382800) else Color(0xFFFFFBFF)
```

Comment:
@QuocNguyen-w3w if we use the semantic name of colors from our design library they can be easily overridden